hilti/toolchain/src/config.cc.in

```diff
@@ -170,3 +170,56 @@ Configuration& hilti::configuration() {
     static Configuration singleton;
     return singleton;
 }
+
+namespace std {
+template<>
+struct hash<hilti::Configuration> {
+    std::size_t operator()(const hilti::Configuration& config) const {
+        std::hash<std::string> hasher;
+
+        std::size_t hash = 0;
+
+        hash = rt::hashCombine(hash, hasher(config.build_directory));
+        hash = rt::hashCombine(hash, hasher(config.compiler_id));
+        hash = rt::hashCombine(hash, hasher(config.cxx));
+        hash = rt::hashCombine(hash, hasher(config.distbase));
+        for ( const auto& path : config.hilti_library_paths )
```

Comment:
How about providing a `hashCombine()` overloaded that takes vector of strings and paths; then we could skip all the for-loops here, which would it make it more readable and easier to maintain/check.

Reply:
I'd prefer to only work on combining hashes in `hashCombine`, and since in addition to hashing strings here we also need to hash `vector`s of `filesystem::path` and `vector`s of `std::string`s this would actually make this pretty complicated.

Instead I added functionality to hash `vector<T>` and `filesystem::path` here which then allows us to simplify the code here somewhat. WDYT?